editor/src/components/canvas/controls/select-mode/grid-gap-control-component.tsx

```diff
@@ -0,0 +1,477 @@
+    ) ?? false
+
+  const grid = useGridData([selectedElement]).at(0)
+
+  const activeDraggingAxis = useEditorState(
+    Substores.canvas,
+    (store) =>
+      store.editor.canvas.interactionSession?.activeControl.type === 'GRID_GAP_HANDLE'
+        ? store.editor.canvas.interactionSession?.activeControl.axis
+        : null,
+    'GridGapControl isDragging',
+  )
+
+  const canvasOffset = useRefEditorState((store) => store.editor.canvas.roundedCanvasOffset)
+
+  const axisMouseDownHandler = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>, axis: Axis) => {
+      startInteraction(e, dispatch, canvasOffset.current, scale, axis)
+    },
+    [canvasOffset, dispatch, scale],
+  )
+  const rowMouseDownHandler = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => axisMouseDownHandler(e, 'row'),
+    [axisMouseDownHandler],
+  )
+
+  const columnMouseDownHandler = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => axisMouseDownHandler(e, 'column'),
+    [axisMouseDownHandler],
+  )
+
+  const [hoveredAxis, setHoveredAxis] = React.useState<'row' | 'column'>('row')
+  const onMouseOverRow = React.useCallback(() => setHoveredAxis('row'), [])
+  const onMouseOverColumn = React.useCallback(() => setHoveredAxis('column'), [])
+
+  const gridGap = useEditorState(
+    Substores.metadata,
+    (store) => maybeGridGapData(store.editor.jsxMetadata, selectedElement),
+    'GridGapControlComponent gridGap',
+  )
+
+  if (grid == null || gridGap == null) {
+    return null
+  }
+
+  return (
+    <CanvasOffsetWrapper>
+      <GridPaddingOutlineForDimension
+        grid={grid}
+        dimension={'rows'}
+        onMouseDown={rowMouseDownHandler}
+        beingDragged={activeDraggingAxis === 'row'}
+        onMouseOver={onMouseOverRow}
+        zIndexPriority={hoveredAxis === 'row' ? true : false}
+        gridGap={gridGap.row}
+        elementHovered={elementHovered}
+      />
+      <GridPaddingOutlineForDimension
+        grid={grid}
+        dimension={'columns'}
+        onMouseDown={columnMouseDownHandler}
+        beingDragged={activeDraggingAxis === 'column'}
+        onMouseOver={onMouseOverColumn}
+        zIndexPriority={hoveredAxis === 'column' ? true : false}
+        gridGap={gridGap.column}
+        elementHovered={elementHovered}
+      />
+    </CanvasOffsetWrapper>
+  )
+})
+
+const GridPaddingOutlineForDimension = (props: {
+  grid: GridData
+  dimension: 'rows' | 'columns'
+  onMouseDown: (e: React.MouseEvent<HTMLDivElement>) => void
+  beingDragged: boolean
+  onMouseOver: () => void
+  zIndexPriority: boolean
+  gridGap: CSSNumberWithRenderedValue
+  elementHovered: boolean
+}) => {
+  const {
+    grid,
+    gridGap,
+    dimension,
+    onMouseDown,
+    beingDragged,
+    onMouseOver,
+    zIndexPriority,
+    elementHovered,
+  } = props
+
+  let style: CSSProperties = {
+    ...getStyleMatchingTargetGrid(grid),
+    zIndex: zIndexPriority ? 1 : 0,
```

Comment:
I believe here as well might be worth leaving it
```suggestion
    zIndex: zIndexPriority ? 1 : undefined,
```